tests/models/test_anthropic.py

```diff
@@ -541,3 +543,34 @@ async def my_tool(first: str, second: str) -> int:
         assert result.is_complete
         assert result.usage() == snapshot(Usage(requests=2, request_tokens=20, response_tokens=5, total_tokens=25))
         assert tool_called
+
+
+@pytest.mark.vcr()
+async def test_image_url_input(allow_model_requests: None, anthropic_api_key: str):
+    m = AnthropicModel('claude-3-5-haiku-latest', api_key=anthropic_api_key)
+    agent = Agent(m)
+
+    result = await agent.run(
+        [
+            'What is this vegetable?',
+            ImageUrl(url='https://t3.ftcdn.net/jpg/00/85/79/92/360_F_85799278_0BBGV9OAdQDTLnKwAPBCcg1J7QtiieJY.jpg'),
+        ]
+    )
+    assert result.data == snapshot("""\
+This is a potato. It's a yellow-skinned potato with a somewhat oblong or oval shape. The surface is covered in small eyes or dimples, which is typical of potato skin. The color is a golden-yellow, and the potato appears to be clean and fresh, photographed against a white background.
+
+Potatoes are root vegetables that are staple foods in many cuisines around the world. They can be prepared in numerous ways such as boiling, baking, roasting, frying, or mashing. This particular potato looks like it could be a Yukon Gold or a similar yellow-fleshed variety.\
+""")
+
+
+@pytest.mark.parametrize('media_type', ('audio/wav', 'audio/mpeg'))
+async def test_audio_as_binary_content_input(allow_model_requests: None, media_type: str):
```

Comment:
We should probably make it so that any test that uses `allow_model_requests` fixture needs `@pytest.mark.vcr`. I understand in this case that you don't need to use `vcr` because you are using a mock_client, but I think this is a problem. I think the best solution is to make it so that using a `mock_client` doesn't require you to allow model requests. This doesn't need to be changed now but I'm just pointing it out.